include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -675,22 +676,23 @@ struct __parallel_radix_sort_iteration
 //-----------------------------------------------------------------------
 // radix sort: main function
 //-----------------------------------------------------------------------
-template <bool __is_ascending, typename _Range, typename _ExecutionPolicy>
+template <bool __is_ascending, typename _Range, typename _ExecutionPolicy, typename _Proj>
 auto
-__parallel_radix_sort(_ExecutionPolicy&& __exec, _Range&& __in_rng)
+__parallel_radix_sort(_ExecutionPolicy&& __exec, _Range&& __in_rng, _Proj __proj)
 {
     const ::std::size_t __n = __in_rng.size();
     assert(__n > 1);
 
     // types
     using _DecExecutionPolicy = __decay_t<_ExecutionPolicy>;
-    using _T = oneapi::dpl::__internal::__value_t<_Range>;
+    using _ValueT = oneapi::dpl::__internal::__value_t<_Range>;
+    using _KeyT = oneapi::dpl::__internal::__key_t<_Proj, _Range>;
 
     // radix bits represent number of processed bits in each value during one iteration
     constexpr ::std::uint32_t __radix_bits = 4;
 
     sycl::buffer<::std::uint32_t, 1> __tmp_buf(sycl::range<1>(0));
```

Comment:
I believe it's better for us to leave a comment why we are creating zero-length buffer.